mtags/src/main/scala-2/scala/meta/internal/pc/TypeCheckCompilationUnit.scala

```diff
@@ -0,0 +1,32 @@
+package scala.meta.internal.pc
+
+import scala.tools.nsc.reporters.StoreReporter
+
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.VirtualFileParams
+
+class TypeCheckCompilationUnit(
+    val compiler: MetalsGlobal,
+    val params: VirtualFileParams
+) {
+  import compiler._
+  val unit: RichCompilationUnit = addCompilationUnit(
+    code = params.text(),
+    filename = params.uri().toString,
+    cursor = None
+  )
+  val offset: Int = params match {
+    case p: OffsetParams => p.offset()
+    case _: VirtualFileParams => 0
+  }
+  val pos: Position = unit.position(offset)
+  lazy val text = unit.source.content
+
+  private val previousReporter = reporter
+
+  val console: StoreReporter = storeReporterConstructor(settings)
```

Comment:
```suggestion
  val diagnoticsReporter: StoreReporter = storeReporterConstructor(settings)
```
maybe? Since this doesn't seem to have anything to do with console?